### PR TITLE
Add single web view discussion

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14161
affects: Student, Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/g7cedns) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fg7cedns)
teacher: [Link](https://rebrand.ly/3vkrqps) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F3vkrqps)
parent: [Link](https://rebrand.ly/uxrftje) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fuxrftje)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher

[MBL-14161](https://instructure.atlassian.net/browse/MBL-14161)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/rn/Teacher/ios/Teacher/AppDelegate+Routes.swift:24:11: Function Body Length Violation: Function body should span 100 lines or less excluding comments and whitespace: currently spans 103 lines (function_body_length)
    Linting 'SpeedGraderUITests.swift' (35/39)
    Linting 'SpeedGrader.swift' (36/39)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Discussions/APIDiscussion.swift [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    }
+
+    public var path: String {
+        "\(context.pathComponent)/discussion_topics/\(topicID)/view"
+    }
+
+    public var query: [APIQueryItem] {
+        var query: [APIQueryItem] = []
+        if includeNewEntries {
+            query.append(.value("include_new_entries", "1"))
```
Comment (armstrongnate) [inline Core/Core/Discussions/APIDiscussion.swift:333]: FYI there is a `.bool` APIQueryItem.

File: Core/Core/Discussions/DiscussionDetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+            publishedIcon.tintColor = .named(.textSuccess)
+            publishedLabel.text = NSLocalizedString("Published", bundle: .core, comment: "")
+            publishedLabel.textColor = .named(.textSuccess)
+        } else {
+            publishedIcon.image = .icon(.no, .solid)
+            publishedIcon.tintColor = .named(.textDark)
+            publishedLabel.text = NSLocalizedString("Unpublished", bundle: .core, comment: "")
+            publishedLabel.textColor = .named(.textDark)
+        }
+        publishedView.isHidden = !Bundle.main.isTeacherApp
```
Comment (armstrongnate) [inline Core/Core/Discussions/DiscussionDetailsViewController.swift:170]: Suggestion: `env.app == .teacher`

**Comment by armstrongnate** [comment]
I know there’s a weird edge case with group discussions that we handle in RN that I don’t see handled here (not expecting it to be handled yet). I can make a separate ticket for it so we don’t forget.

**Pull request merged**